src/components/TextInput/TextInput.jsx

```diff
@@ -57,21 +57,24 @@ class TextInput extends React.Component {
   }
 
   render() {
+    let ariaDescribedBy = this.props.ariaDescribedBy;
     // Calculate error state.
     let errorSpan = '';
     let maxCharacters;
-    let errorSpanId = undefined;
     let inputErrorClass = undefined;
     let labelErrorClass = undefined;
     if (this.props.errorMessage) {
-      errorSpanId = `${this.inputId}-error-message`;
+      const errorSpanId = `${this.inputId}-error-message`;
       errorSpan = (
         <span className="usa-input-error-message" role="alert" id={errorSpanId}>
           <span className="sr-only">Error</span> {this.props.errorMessage}
         </span>
       );
       inputErrorClass = 'usa-input-error';
       labelErrorClass = 'usa-input-error-label';
+      ariaDescribedBy = ariaDescribedBy
+        ? `${ariaDescribedBy} ${errorSpanId}`
+        : errorSpanId;
```

Comment:
I'm not sure I understand this. `ariaDescribedBy` could potentially be any DOM element. Appending the `${errorSpanId}` tightly couples this component to implementation details outside of this component.

Under what circumstances should the `aria-describedby` property be different than the error message rendered by this component? Or am I missing something?

Reply:
The reply in the root thread has the background story, but there was a case where we had additional description that we wanted to tie for accessibility reasons into a TextInput. 

From what I can gather on the aria-describedby documentation, it can take in multiple identifies separated by spaces. I wanted to have any ariaDescribedBy proceed the errorSpanId because the error message/information would be read last, after all the additional descriptions.

Please let me know if any of that doesn't makes sense, or if you have any recommendations!

Reply:
> it can take in multiple identifies separated by spaces.

Ahhhh, so it does, [look at that](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute#value). TIL. :smile: That's not misleading at all. :stuck_out_tongue_closed_eyes: